### PR TITLE
remove: delete .gitignore file if empty

### LINKED ISSUE #4962: remove: delete .gitignore file if empty
Just like `dvc remove` deletes dvc.yaml and dvc.lock if they're empty after the removal, should it do the same to .gitignore? Currently it's left there even if empty, which shows up as a change in git (so it's pretty obvious).

### PR DESCRIPTION
**Pull request by devramx**
modified behavior to unlink empty .gitignore files.
added corresponding unit & func test cases.
updated failing test cases.
no failing tests

Fixes #4962

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/utils/__init__.py [hunk 1]
```diff
@@ -9,6 +9,10 @@ def get_gitignore_content():
         return gitignore.read().splitlines()
 
 
+def gitignore_exists():
+    return os.path.exists(Git.GITIGNORE)
+
```
Comment (efiop) [inline tests/utils/__init__.py:14]: Let's just inline it, it is not saving us any code or readablility.

File: tests/unit/scm/test_git.py [hunk 2]
```diff
@@ -279,3 +279,36 @@ def test_list_all_commits(tmp_dir, scm):
     scm.set_ref("refs/foo/bar", rev_c)
 
     assert {rev_a, rev_b} == set(scm.list_all_commits())
+
+
+def test_ignore_remove_empty(tmp_dir, scm):
+    from dvc.scm.git import Git
+
+    git_ = Git(os.fspath(tmp_dir))
```
Comment (efiop) [inline tests/unit/scm/test_git.py:287]: Seems like it is unused?

File: tests/func/test_remove.py [hunk 3]
```diff
@@ -69,3 +69,35 @@ def test_cmd_remove(tmp_dir, dvc):
...
+
+def test_cmd_remove_gitignore_single_stage(tmp_dir, scm, dvc, run_copy):
+    stage = dvc.run(
+        name="my", cmd='echo "hello" > out', deps=[], outs=["out"],
+    )
+
+    assert (tmp_dir / ".gitignore").exists(), ".gitignore does not exist"
+
+    assert main(["remove", stage.addressing]) == 0
+    assert not (tmp_dir / stage.relpath).exists(), "Pipeline file does exists"
```
Comment (efiop) [inline tests/func/test_remove.py:82]: It is pretty self explanatory, we tend to not use assert messages instead trying to make the code self-explanatory, so for consistency let's just:
```suggestion
    assert not (tmp_dir / stage.relpath).exists()
```

**Review by efiop — APPROVED** [review]
Looks great! Just a few minor comments to address.

**Review by efiop — APPROVED** [review]
Looks good! Thank you!

**Comment by efiop** [comment]
@devramx One more thing though. Let's actually document the behavior in https://dvc.org/doc/command-reference/remove . Mind creating an issue on dvc.org or creating a PR for it?

**Comment by devramx** [comment]
Ok. Will create PR in dvc.org.

**Pull request merged**